pkg/process/checks/net.go

```diff
@@ -115,15 +119,20 @@ func (c *ConnectionsCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.
 	if err != nil {
 		log.Warnf("error collecting processes for proxy filter: %s", err)
 	} else {
-		dockerproxy.NewFilter(procs).Filter(conns)
+		for _, p := range procs {
+			c.dockerFilter.Extract(p)
+			c.serviceExtractor.Extract(p)
```

Comment:
The above makes sense, but maybe I'm missing something from reading the existing implementation of how the DockerProxy filter is implemented. I see the [LoadProxies()](https://github.com/DataDog/datadog-agent/blob/4aa9869efe421349048e45f836c168db79fe1437/pkg/process/dockerproxy/filter.go#L39-L60) method iterates across all processes on every connection check. I was replicating that iteration logic instead of encapsulating it inside the filter. Maybe it was a premature optimization to avoid looping across the same processes, which ties back to your later [point](https://github.com/DataDog/datadog-agent/pull/14287#discussion_r1029838459). I'm starting to hesitate using this extractor interface at the moment as I want to flush out more use cases. I can see a future where this type of process data extraction can occur outside of a single check and made available across checks (like tagging)

Reply:
> but maybe I'm missing something from reading the existing implementation of how the DockerProxy filter is implemented.

you're actually right. This `LoadProxies` could actually take as an argument the same "subset" of processes (including only PIDs that are present in the `conns` slice).

> I can see a future where this type of process data extraction can occur outside of a single check and made available across checks (like tagging)

Yup, I agree that in _that_ context it would be useful to have an interface because maybe you'll somehow _register_ a `metadata.Extractor` to a global process check of some sort, and this global thing would execute these `Extract` methods as callbacks or something like that.